modules/metastorage-server/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -179,25 +179,30 @@ public void start() {
 
         ColumnFamilyOptions indexFamilyOptions = new ColumnFamilyOptions(indexOptions);
 
-        List<ColumnFamilyDescriptor> descriptors = Arrays.asList(
+        return List.of(
                 new ColumnFamilyDescriptor(DATA.nameAsBytes(), dataFamilyOptions),
                 new ColumnFamilyDescriptor(INDEX.nameAsBytes(), indexFamilyOptions)
         );
+    }
 
-        var handles = new ArrayList<ColumnFamilyHandle>();
+    private void recreateDb() throws RocksDBException {
+        destroyRocksDb();
 
-        try {
-            // Delete existing data, relying on the raft's snapshot and log playback
-            destroyRocksDb();
+        List<ColumnFamilyDescriptor> descriptors = cfDescriptors();
 
-            this.db = RocksDB.open(options, dbPath.toAbsolutePath().toString(), descriptors, handles);
-        } catch (RocksDBException e) {
-            throw new IgniteInternalException("Failed to start the storage", e);
-        }
+        var handles = new ArrayList<ColumnFamilyHandle>(descriptors.size());
+
+        options = new DBOptions()
+                .setCreateMissingColumnFamilies(true)
+                .setCreateIfMissing(true);
+
+        db = RocksDB.open(options, dbPath.toAbsolutePath().toString(), descriptors, handles);
+
+        data = ColumnFamily.createExisting(db, handles.get(0));
```

Comment:
`recreateDb` is called inside `start`, but here you use `createExisting`. What if column family doesn't exist?

Reply:
Oh, I see, you use create missing column families